autoload/OmniSharp/locations.vim

```diff
@@ -23,14 +23,20 @@ function! OmniSharp#locations#Navigate(location, noautocmds) abort
   endif
 endfunction
 
-function! OmniSharp#locations#Parse(quickfixes) abort
+function! s:DefaultFixup(quickfix) abort
+  return a:quickfix
+endfunction
+
+function! OmniSharp#locations#Parse(quickfixes, ...) abort
+  let Fixup = get(a:, 1, function("s:DefaultFixup"))
   let locations = []
-  let overrides = get(g:, 'OmniSharp_diagnostic_overrides', {})
   for quickfix in a:quickfixes
-    let text = get(quickfix, 'Text', get(quickfix, 'Message', ''))
-    if get(g:, 'OmniSharp_diagnostic_showid') && has_key(quickfix, 'Id')
-      let text = quickfix.Id . ': ' . text
+    let quickfix = Fixup(quickfix)
```

Comment:
I don't think we need the default function, what about this:

```vim
let quickfix = a:0 ? a:1(quickfix) : quickfix
```

Reply:
I'm really new to vimscript (this is my first ever writing it) so this is super helpful thanks! I made the proposed change and it works.

Reply:
And you're already passing around funcref callbacks 😄 Great stuff. The language definitely has its quirks but I quite like it.